docs/schema_registry_tutorial.rst

```diff
@@ -0,0 +1,505 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In our example of the Payment schema, let's say now some applications are sending additional information for each payment, e.g., a field `region` that represents the place of sale.
+Consider the `Payment2a schema <https://github.com/confluentinc/examples/blob/DEVX-380/clients/avro/src/main/resources/avro/io/confluent/examples/clients/basicavro/Payment2a.avsc>`_:
+
+.. sourcecode:: json
+
+   $ cat src/main/resources/avro/io/confluent/examples/clients/basicavro/Payment2a.avsc
+   [
+   {"namespace": "io.confluent.examples.clients.basicavro",
+    "type": "record",
+    "name": "Payment",
+    "fields": [
+        {"name": "id", "type": "string"},
+        {"name": "amount", "type": "double"},
+        {"name": "region", "type": "string"}
+    ]
+   }
+   ]
+
+Before proceeding, think about whether this schema is backward compatible.
+Specifically, ask yourself whether a consumer can use this schema to read data written by producers using the older schema without the `region` field?
+The answer is no.
+Consumers will fail reading data with the older schema, because the older schema does not have the `region` field, so it is not backward compatible.
+
+Confluent provides a `Schema Registry Maven Plugin <https://docs.confluent.io/current/schema-registry/docs/maven-plugin.html#sr-maven-plugin>`_ with which you can check compatibility in development.
+Our sample `pom.xml <https://github.com/confluentinc/examples/blob/5.0.0-post/clients/avro/pom.xml#L84-L99>`_ uses this plugin to enable compatibility checks.
+
+.. sourcecode:: xml
+
+      <plugin>
+          <groupId>io.confluent</groupId>
+          <artifactId>kafka-schema-registry-maven-plugin</artifactId>
+          <version>5.0.0</version>
+          <configuration>
+              <schemaRegistryUrls>
+                  <param>http://localhost:8081</param>
+              </schemaRegistryUrls>
+              <subjects>
+                  <transactions-value>src/main/resources/avro/io/confluent/examples/clients/basicavro/Payment2a.avsc</transactions-value>
+              </subjects>
+          </configuration>
+          <goals>
+              <goal>test-compatibility</goal>
+          </goals>
+      </plugin>
+
+It is currently configured to check compatibility of the new `Payment2a` schema for the `transactions-value` subject in |sr|.
+Run the compatibility check and verify that it fails:
+
+.. sourcecode:: bash
+
+   $ mvn io.confluent:kafka-schema-registry-maven-plugin:5.0.0:test-compatibility
+   ....
+   [ERROR] Schema /Users/yeva/git/examples/clients/avro/src/main/resources/avro/io/confluent/examples/clients/basicavro/Payment2a.avsc is not compatible with subject(transactions-value)
+   ....
+
+There is alternative way to manually try to register the schema to |sr|, which is useful for non-Java clients.
+You can try registering the new schema `Payment2a` directly, but |sr| rejects it, with an error message that it is incompatible.
+
+.. sourcecode:: bash
+
```

Comment:
Might want to add a note (somewhere) about escaping JSON in JSON..

Reply:
Good point